### ISSUE TITLE
Bug: The `Reflect.getOwnPropertyNames` still doesn't exist

### ISSUE THREAD

**Issue by AnnAngela** [eslint/eslint#16305]
### Environment

Node version: v16.17.0
npm version: v8.18.0
Local ESLint version: v8.23.0 (Currently used)
Global ESLint version: v8.23.0
Operating System: linux 5.4.0-1089-azure

### What parser are you using?

Default (Espree)

### What did you do?

<details>
<summary>Configuration</summary>

```
{
    "env": {
        "browser": true,
        "jquery": true,
        "es6": true,
        "es2020": true,
        "es2021": true
    },
    "parserOptions": {
        "ecmaVersion": "latest"
    },
    "extends": "eslint:recommended",
    "rules": {
        "prefer-reflect": "error"
    }
}
```
</details>

Code:

```js
        function getOwnPropertyNamesLength(obj) {
            return Object.getOwnPropertyNames(obj).length;
        }
```

### What did you expect to happen?

Due to the function `Reflect.getOwnPropertyNames` still doesn't exist (I can't even find it in caniuse.com), the rule `prefer-reflect` should not consider about it, so does the document.

### What actually happened?

The rule complain that "Avoid using Object.getOwnPropertyNames, instead use Reflect.getOwnPropertyNames."

### Participation

- [X] I am willing to submit a pull request for this issue.

### Additional comments

I knew that #7075 is about it and closed, but I met the problem again, that's why I submit this issue.

**Comment by mdjermanovic** [comment]
Hi @AnnAngela, thanks for the issue!

You're right, this looks like an oversight. But, this rule is [deprecated](https://eslint.org/docs/latest/rules/prefer-reflect) so we are not making any further changes to it.

To avoid this problem, you can add `"getOwnPropertyNames"` to [`exceptions`](https://eslint.org/docs/latest/rules/prefer-reflect#exceptions):

```json
{
    "rules": {
        "prefer-reflect": ["error", { "exceptions": ["getOwnPropertyNames"] }]
    }
}
```

[Playground example](https://eslint.org/play/#eyJ0ZXh0IjoiLyogZXNsaW50IFwicHJlZmVyLXJlZmxlY3RcIjogW1wiZXJyb3JcIiwgeyBcImV4Y2VwdGlvbnNcIjogW1wiZ2V0T3duUHJvcGVydHlOYW1lc1wiXSB9XSAqL1xuXG5mdW5jdGlvbiBnZXRPd25Qcm9wZXJ0eU5hbWVzTGVuZ3RoKG9iaikge1xuICAgIHJldHVybiBPYmplY3QuZ2V0T3duUHJvcGVydHlOYW1lcyhvYmopLmxlbmd0aDsgLy8gbm8gZXJyb3Jcbn07Iiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjoibGF0ZXN0Iiwic291cmNlVHlwZSI6InNjcmlwdCIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6e30sImVudiI6eyJlczYiOnRydWV9fX0=)

**Comment by AnnAngela** [comment]
@mdjermanovic But at least I think it should be disclosed in documents. And if I make a PR about this, will it be able to be accepted?

**Comment by mdjermanovic** [comment]
> @mdjermanovic But at least I think it should be disclosed in documents. And if I make a PR about this, will it be able to be accepted?

Yes, we can accept documentation updates for this rule. PR is welcome!

**Comment by AnnAngela** [comment]
@mdjermanovic can I directly remove the codes about `getOwnPropertyNames` in the rule file? so the rule will work fine and won't break peoples' work anymore.

**Comment by ljharb** [comment]
The appropriate Reflect method you should be using here is called `Reflect.ownKeys`.

**Comment by mdjermanovic** [comment]
@AnnAngela per our [rule deprecation policy](https://eslint.org/docs/latest/user-guide/rule-deprecation), we're not making any changes to deprecated rules. In this case, we can make an exception to describe a workaround in the docs, but we're definitely not making any changes in the code.

@ljharb `Reflect.ownKeys` includes symbols, so it's not the same as `Object.getOwnPropertyNames `?

**Comment by ljharb** [comment]
@mdjermanovic that's a fair point, but it's just `Reflect.ownKeys(x).filter(x => typeof x === 'string')` ¯\\\_(ツ)_/¯

**Comment by mdjermanovic** [comment]
> that's a fair point, but it's just `Reflect.ownKeys(x).filter(x => typeof x === 'string')` ¯\_(ツ)_/¯

That would work, but this rule wasn't intended to suggest replacements of such complexity, so I think it's fine to add a note in the docs that `Object.getOwnPropertyNames` was added by mistake.

**Comment by AnnAngela** [comment]
@mdjermanovic OK, I will make a PR to change the documents only.
